R/pkg/R/LM.R

```diff
@@ -112,40 +115,31 @@ predict.sparkLM <- function(model, newData) {
 #' > Model:
 #' > y ~ x1 + x2 + x3
 #' }
-summaryObj.sparkLM <- function(model) {
-  if (class(model) != "sparkLM") {
+summary.sparkLM <- function(object, ...) {
+  if (class(object) != "sparkLM") {
     stop("model must be a sparkLM model object.")
   }
   rawSummary <- SparkR:::callJStatic("com.Alteryx.sparkGLM.LM",
                                      "summaryArray",
                                      model$jobj)
-  summaryOut <- list()
-  summaryOut$model <- rawSummary[[1]]
-  summaryOut$coefficients <- rawSummary[[2]]
-  summaryOut$RSE <- rawSummary[[3]]
-  summaryOut$R2 <- rawSummary[[4]]
-  summaryOut$Fstat <- rawSummary[[5]]
-  summaryOut$printSummary <- function() {
-    cat("\nModel:\n")
-    cat(summaryOut$model)
-    cat("\n")
-    cat("\nCoefficients:\n")
-    cat(summaryOut$coefficients, "\n")
-    cat("\n")
-    for(i in c("RSE", "R2", "Fstat")) {
-      cat(summaryOut[[i]], "\n")
-    }
-  }
-  summaryOut
+  names(rawSummary) <- c('model', 'coefficients', 'RSE', 'R2', 'Fstat')
+  class(rawSummary) <- 'summary.sparkLM'
+  return(rawSummary)
```

Comment:
With consistency in mind, you might want to look at the names of the objects in `summary.lm`. While you may not have all these objects, I think it might make sense to keep the names of those in there consistent. Just a thought.

```
mod = lm(mpg ~ wt, data = mtcars)
names(summary(mod))
 [1] "call"          "terms"         "residuals"     "coefficients"  "aliased"      
 [6] "sigma"         "df"            "r.squared"     "adj.r.squared" "fstatistic"   
[11] "cov.unscaled" 
```